net-mgmt/telegraf/src/opnsense/service/templates/OPNsense/Telegraf/telegraf.conf

```
@@ -127,6 +127,12 @@
 {%   if helpers.exists('OPNsense.telegraf.output.elastic_url') and OPNsense.telegraf.output.elastic_url != '' %}
   urls = ["{{ OPNsense.telegraf.output.elastic_url }}"]
 {%   endif %}
+{%   if helpers.exists('OPNsense.telegraf.output.elastic_username') and OPNsense.telegraf.output.elastic_username != '' %}
+  username = "{{ OPNsense.telegraf.output.elastic_username }}"
+{%   endif %}
+{%   if helpers.exists('OPNsense.telegraf.output.elastic_password') and OPNsense.telegraf.output.elastic_password != '' %}
+  password = "{{ OPNsense.telegraf.output.elastic_password }}"
```

```suggestion
{%   if helpers.exists('OPNsense.telegraf.output.elastic_password') and OPNsense.telegraf.output.elastic_password != '' %}
  username = "{{ OPNsense.telegraf.output.elastic_username }}"
  password = "{{ OPNsense.telegraf.output.elastic_password }}"
{%   endif %}
```

as previously stated I would prefer this approach (only works with username and password set)

---

Yes, I agree. How I would accomplish this I honestly did not compute. I see what you did and I thank you. 

 I was in the process of trying to clean up the previous commit and to figure out this "split PR" thing, and then I would "take it from there". Now we are here :)

Probably somewhere I think there should be an option or two in regards to TLS certificates. I'll come back to you in a jiffy to see if this is true.